mbuild/formats/lammpsdata.py

```diff
@@ -651,9 +661,22 @@ def _get_bond_types(
             )
         )
     )
+
+    lx = len(atom_types)
+    bond_sets = [x[2] for _, x in unique_bond_types.items()]
+    ordered_bond_sets = [
+        (atom_types[x], atom_types[y])
+        for x in range(lx)
+        for y in range(x, lx)
+        if (atom_types[x], atom_types[y]) in bond_sets
+    ]
     unique_bond_types = OrderedDict(
-        [(y, x + 1) for x, y in unique_bond_types.items()]
+        [
+            (unique_bond_types[bond_sets.index(x)], i + 1)
+            for i, x in enumerate(ordered_bond_sets)
```

Comment:
```suggestion
            for i, x in enumerate(ordered_bond_tuples)
```
See line 667 comment.